src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -761,19 +761,34 @@ def config(self, spec: RGWSpec, rgw_id: str) -> None:  # type: ignore
         assert self.TYPE == spec.service_type
 
         # set rgw_realm and rgw_zone, if present
-        if spec.rgw_realm:
+        if spec.rgw_zone:
+            if not spec.rgw_realm:
+                raise OrchestratorError(f'{spec} has a zone {spec.rgw_zone}, but no realm')
+            ret, out, err = self.mgr.check_mon_command({
+                'prefix': 'config set',
+                'who': f"{utils.name_to_config_section('rgw')}.{spec.service_id}",
+                'name': 'rgw_zone',
+                'value': spec.rgw_zone,
+            })
             ret, out, err = self.mgr.check_mon_command({
                 'prefix': 'config set',
                 'who': f"{utils.name_to_config_section('rgw')}.{spec.service_id}",
                 'name': 'rgw_realm',
                 'value': spec.rgw_realm,
             })
-        if spec.rgw_zone:
+        else:
+            # explicitly set zone AND zonegroup to 'default'
             ret, out, err = self.mgr.check_mon_command({
                 'prefix': 'config set',
                 'who': f"{utils.name_to_config_section('rgw')}.{spec.service_id}",
                 'name': 'rgw_zone',
-                'value': spec.rgw_zone,
+                'value': 'default',
+            })
+            ret, out, err = self.mgr.check_mon_command({
+                'prefix': 'config set',
+                'who': f"{utils.name_to_config_section('rgw')}.{spec.service_id}",
+                'name': 'rgw_zonegroup',
+                'value': 'default',
```

Comment:
@yehudasa We need a new rgw_zonegroup entry in the spec, right?

Reply:
@sebastian-philipp Sounds like it. I know that @liewegas was trying to avoid using zonegroups at all.

Reply:
What's the deal with zonegroups? We have https://tracker.ceph.com/issues/48340 and knowing why we should or shouldn't add zonegroups would be important

Reply:
zonegroups originally defined the set of zones that should be replicated symmetrically (for DR purposes). Currently (having bucket granularity sync), they define groups of zones that can sync from each other. Some of the configuration is defined in the zonegroup level as it applies to all the zones in that group. Hiding the zonegroup would imply a single zonegroup in the realm that all the zones belong to.